src/components/shapeComponents/SetDefinitionForm.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div v-for="(included, clauseIndex) of clauses">
+    <li class="rule-container">
+      <Button icon="pi pi-times" class="p-button-rounded p-button-danger p-button-text" @click="removeClause(clauseIndex)" />
+      <Dropdown v-model="included.include" :options="includeOptions" option-label="name" option-value="value" placeholder="Include or exclude" />
+      <EntityAutocomplete :ttAlias="included.concept" />
+      <Dropdown
+        v-model="included.concept.includeSubtypes"
+        :options="includeSubtypesOptions"
+        option-label="name"
+        option-value="value"
+        placeholder="Include subtypes"
+      />
+    </li>
+
+    <div v-for="(refinement, refinementIndex) of included.refinements">
+      <ul class="refinement-container">
+        <Button icon="pi pi-times" class="p-button-rounded p-button-danger p-button-text" @click="removeRefinement(clauseIndex, refinementIndex)" />
+        <EntityAutocomplete
+          :ttAlias="refinement.property"
+          :parentClauseIri="included.concept['@id']"
+          :get-suggestions-method="getAllowablePropertySuggestions"
+        />
+        <Dropdown
+          v-model="refinement.property.includeSubtypes"
+          :options="includeSubtypesOptions"
+          option-label="name"
+          option-value="value"
+          placeholder="Include subtypes"
+        />
+        <i class="icon pi pi-arrow-right" />
+        <EntityAutocomplete :ttAlias="refinement.is" :parentClauseIri="refinement.property['@id']" :get-suggestions-method="getAllowableRangeSuggestions" />
+        <Dropdown
+          v-model="refinement.is.includeSubtypes"
+          :options="includeSubtypesOptions"
+          option-label="name"
+          option-value="value"
+          placeholder="Include subtypes"
+        />
+      </ul>
+    </div>
+    <ul class="refinement-button">
+      <Button icon="pi pi-cog" label="Add refinement" class="p-button-warning" @click="addRefinement(clauseIndex)" />
+    </ul>
+    <div class="rule-button">
+      <Button v-if="clauseIndex === clauses.length - 1" icon="pi pi-plus" label="Add rule" class="p-button-success" @click="addConcept" />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { SetQueryObject, TTAlias, Refinement } from "im-library/dist/types/interfaces/Interfaces";
+import { PropType } from "vue";
+import EntityAutocomplete from "./setDefinition/EntityAutocomplete.vue";
```

Comment:
Does the functionality of this differ greatly from the shapeComponent version of EntityAutocomplete? Can we combine them into 1?